### PR TITLE
Feature/add reduce and start key to mock

### PR DESCRIPTION
**Pull request by Reettaphant**

### CODE REVIEW

File: test/qouch-mock.js [hunk 1]
```diff
@@ -169,10 +169,84 @@ function qouchMockFactory ( docs, designDocPaths ) {
...
+            if ( row1.key < row2.key ) {
+              return 1;
+            }
+            if ( row1.key > row2.key ) {
+              return -1;
+            }
+            return 0;
+          }
+
+          for ( i = 0; i < row1.key.length; i++ ) {
```
Comment (jamesallardice) [inline test/qouch-mock.js]: Is `i` declared anywhere?

File: test/qouch-mock.js [hunk 2]
```diff
@@ -169,10 +169,84 @@ function qouchMockFactory ( docs, designDocPaths ) {
...
+
+      var keys = matchedRows.map(function ( matchedRow ) {
+        return [ matchedRows.key, matchedRows.id ];
+      });
+
+      var values = matchedRows.map(function ( matchedRow ) {
+        return matchedRow.doc;
+      });
+
+      var firstReducedValues = reduceFn(keys.slice(0, matchedRows.length/2), values.slice(0, matchedRows.length/2), false);
```
Comment (jamesallardice) [inline test/qouch-mock.js]: Styleguide: spaces around `/`

File: test/qouch-mock.js [hunk 3]
```diff
@@ -169,10 +169,84 @@ function qouchMockFactory ( docs, designDocPaths ) {
...
+        return [ matchedRows.key, matchedRows.id ];
+      });
+
+      var values = matchedRows.map(function ( matchedRow ) {
+        return matchedRow.doc;
+      });
+
+      var firstReducedValues = reduceFn(keys.slice(0, matchedRows.length / 2), values.slice(0, matchedRows.length / 2), false);
+      var secondReducedValues = reduceFn(keys.slice(matchedRows.length / 2, matchedRows.length), matchedRows.slice(matchedRows.length / 2, matchedRows.length), false);
+      if ( !secondReducedValues ) {
```
Comment (BBB) [inline test/qouch-mock.js]: Under what condition would `secondReducedValues` be falsy here?

Comment (Reettaphant) [inline test/qouch-mock.js]: Under no conditions, I'll fix this.

**Pull request merged**